packages/gatsby-plugin-page-creator/src/__tests__/extract-query.ts

```diff
@@ -0,0 +1,161 @@
+
+describe(`extract query`, () => {
+  describe(`root query string`, () => {
+    it(`basic example`, () => {
+      expect(
+        generateQueryFromString(
+          `Thing`,
+          compatiblePath(`/foo/bar/{Thing.id}.js`)
+        )
+      ).toBe(`{allThing{nodes{id}}}`)
+    })
+
+    it(`works with different file extsnions`, () => {
+      expect(
+        generateQueryFromString(
+          `Thing`,
+          compatiblePath(`/foo/bar/{Thing.id}.tsx`)
+        )
+      ).toBe(`{allThing{nodes{id}}}`)
+    })
+
+    it(`works with arguments arguments`, () => {
+      expect(
+        generateQueryFromString(
+          `{ allThing(filter: { main_url: { nin: [] }}) { ...CollectionPagesQueryFragment } }`,
+          compatiblePath(`/foo/bar/{Thing.id}.js`)
+        )
+      ).toBe(`{ allThing(filter: { main_url: { nin: [] }}) { nodes{id} } }`)
+    })
+
+    it(`supports a special fragment`, () => {
+      expect(
+        generateQueryFromString(
+          `{ allMarkdownRemark {
+        group(field: frontmatter___topic) {
+            ...CollectionPagesQueryFragment
+        }}
+    }`,
+          compatiblePath(`/foo/bar/{MarkdownRemark.frontmatter__topic}.js`)
+        )
+      ).toEqual(`{ allMarkdownRemark {
+        group(field: frontmatter___topic) {
+            nodes{frontmatter{topic},id}
+        }}
+    }`)
+    })
+  })
+
+  describe(`filepath resolution`, () => {
+    it(`basic example`, () => {
+      expect(
+        generateQueryFromString(
+          `Thing`,
+          compatiblePath(`/foo/bar/{Thing.id}.js`)
+        )
+      ).toBe(`{allThing{nodes{id}}}`)
+    })
+
+    it(`always queries id`, () => {
+      expect(
+        generateQueryFromString(
+          `Thing`,
+          compatiblePath(`/foo/bar/{Thing.baz}.js`)
+        )
+      ).toBe(`{allThing{nodes{baz,id}}}`)
+    })
+
+    it(`multiple nodes`, () => {
+      expect(
+        generateQueryFromString(
+          `Thing`,
+          compatiblePath(`/foo/bar/{Thing.id}/{Thing.name}.js`)
+        )
+      ).toBe(`{allThing{nodes{id,name}}}`)
+    })
+
+    it(`nested nodes`, () => {
+      expect(
+        generateQueryFromString(
+          `Thing`,
+          compatiblePath(`/foo/bar/{Thing.id}/{Thing.fields__name}.js`)
+        )
+      ).toBe(`{allThing{nodes{id,fields{name}}}}`)
+    })
+
+    it(`deeply nested nodes`, () => {
+      expect(
+        generateQueryFromString(
+          `Thing`,
+          compatiblePath(`/foo/bar/{Thing.id}/{Thing.fields__name__thing}.js`)
+        )
+      ).toBe(`{allThing{nodes{id,fields{name{thing}}}}}`)
+    })
+
+    it(`supports graphql unions`, () => {
+      expect(
+        generateQueryFromString(
+          `UnionQuery`,
+          compatiblePath(
+            `/foo/bar/{UnionQuery.id}/{UnionQuery.parent__(File)__relativePath}.js`
+          )
+        )
+      ).toBe(`{allUnionQuery{nodes{id,parent{... on File{relativePath}}}}}`)
+    })
+
+    it(`supports nested graphql unions`, () => {
```

Comment:
Isn't this is the exact same test as in line 103 ("supports graphql unions")?

What needs to also work is something like `/foo/bar/{Fruits.parent__(FruitsYaml)__parent__(File)__name}.js`, see:
- GitHub: https://github.com/LekoArts/unified-routes-demo/blob/master/src/pages/fruits/%7BFruits.parent__(FruitsYaml)__parent__(File)__name%7D.js
- Slack: https://gatsbyjs.slack.com/archives/C0137DT736V/p1595837372009500

Reply:
Yup, I messed up :) fixed!